core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -129,14 +134,15 @@ public T acquire(Integer time, TimeUnit unit) {
         mTakeLock.unlock();
       }
     } catch (InterruptedException e) {
-      throw new RuntimeException(e);
+      throw new IOException(e);
```

Comment:
which contract does it break? The way you are doing (catch IOException and log) does not seem correct to me.

An InterruptedException happens when someone interrupts the thread that calls `acquire`. We do not have code does that. When that happens (the thread is interrupted), we should really throw InterruptedException here (similar as how DynamicResource does). If that never happens if the code works properly, RuntimeException is better.